### PR TITLE
[StorageQuota] support account level storage quota

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR adds support of account level storage quota.

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java [hunk 1]
```diff
@@ -37,21 +37,23 @@
...
    * {
    *   "101": {
    *     "1": 1024000000,
    *     "2": 258438456
    *   },
    *   "102": {
    *     "1": 10737418240
-   *   }
+   *   },
+   *   "103: 10737418240
```
Comment (ankagrawal) [inline ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java:49]: nit: missing quotes

File: ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java [hunk 2]
```diff
@@ -37,21 +37,23 @@
...
+   * The key of the top object is the account id and the key of the inner object is the container id.
+   * If there is no inner object, then the value is the storage quota in bytes for this account.
    * The value of the each container id is the storage quota in bytes for this container.
    *
-   * If the targeted container doesn't have a storage quota in this JSON string, it's up to StorageQuotaEnforcer
-   * to decide whether to allow uploads or not.
+   * If the targeted account/container doesn't have a storage quota in this JSON string, it's up to
+   * StorageQuotaEnforcer to decide whether to allow uploads or not.
    */
   @Config(CONTAINER_STORAGE_QUOTA_IN_JSON)
```
Comment (ankagrawal) [inline ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java:58]: nit: rename?

**Review by ankagrawal — APPROVED** [review]
some minor comments. fix it and merge it.

File: ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java]: done.

File: ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-api/src/main/java/com/github/ambry/config/StorageQuotaConfig.java]: done

File: ambry-quota/src/main/java/com/github/ambry/quota/storage/JSONStringStorageQuotaSource.java [hunk 3]
```diff
@@ -82,4 +117,74 @@ public Quota getQuota(QuotaResource quotaResource, QuotaName quotaName) {
...
     // no-op
   }
+
+  @Override
+  public AccountService getAccountService() {
+    return accountService;
+  }
+
+  /**
+   * A helper class to represent a number of a map of string to number. This is used in deserializing json string from
```
Comment (Arun-LinkedIn) [inline ambry-quota/src/main/java/com/github/ambry/quota/storage/JSONStringStorageQuotaSource.java:127]: nit: number 'or' a map of string to number?

**Review by Arun-LinkedIn — APPROVED** [review]
lgtm!

**Pull request merged**